sds/src/scanner/mod.rs

```diff
@@ -439,6 +449,8 @@ struct ScannerContentVisitor<'a, E: Encoding> {
     scanner: &'a Scanner,
     caches: CachePoolGuard<'a>,
     rule_matches: &'a mut Vec<(crate::Path<'static>, Vec<InternalRuleMatch<E>>)>,
+    // Rules that shall be skipped for this scan
+    blocked_rules: &'a Vec<bool>,
```

Comment:
Consider making this optional, so the allocation isn't required in the most frequent case (all false). This might not make a significant different either way though since it's only once per scan.

Reply:
I moved back to a Vec<usize> containing the list of rules. This is simple and should not impact memory nor CPU in most cases.